eodag/plugins/search/data_request_search.py

```diff
@@ -317,10 +317,15 @@ def _convert_result_data(
         """Build EOProducts from provider results"""
         results_entry = self.config.results_entry
         results = result_data[results_entry]
-        normalize_remaining_count = len(results)
+        # postprocess filtering needed when provider does not natively offer
+        # filtering by id
+        if "id" in kwargs:
+            results = [
+                r for r in results if r["productInfo"]["product"] == kwargs["id"]
+            ] or results
+
```

Comment:
these test & filtering should be done after normalizing by checking `products[].properties["id"]`